xpk.py

```diff
@@ -3231,6 +3332,65 @@ def create_cluster_if_necessary(
     )
 
 
+def is_cluster_using_clouddns(args) -> int:
+  """Checks if cluster is using CloudDNS.
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if cluster is using CloudDNS and 1 otherwise.
+  """
+  command = (
+      f'gcloud container clusters describe {args.cluster}'
+      f' --project={args.project} --region={zone_to_region(args.zone)}'
+      ' | grep "clusterDns: CLOUD_DNS" | wc -l'
+  )
+  return_code, cloud_dns_matches = run_command_for_value(
+      command,
+      'Check if Cloud DNS is enabled in cluster describe.',
+      args,
+  )
+  if return_code != 0:
+    xpk_exit(return_code)
+  cloud_dns_matches = int(cloud_dns_matches)
+  if cloud_dns_matches > 0:
+    xpk_print('Cloud DNS is enabled on the cluster, no update needed.')
+    return 0
+  return 1
+
+
+def update_pathways_cluster_if_necessary(args) -> int:
+  """Updates a Pathways cluster to use CloudDNS.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  all_clusters, return_code = get_all_clusters_programmatic(args)
+  if return_code > 0:
+    xpk_print('Listing all clusters failed!')
+    return 1
+  if args.cluster in all_clusters:
```

Comment:
Is it an error case `if args.cluster in all_clusters:` is not true

Reply:
No, it is not an error case. 

if args.cluster in all_clusters: is not true --- it means the cluster is not created and will follow the "create" flow, not the "update" flow. (Hence no error from update perspective.)